internal/resources/policy/kind/network/recipe/allow_all_to_pods_flatten_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package recipe
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipenetworkmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network"
+	policyrecipenetworkcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network/common"
+)
+
+func TestFlattenAllowAllToPods(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		description string
+		input       *policyrecipenetworkmodel.V1alpha1CommonPolicySpecNetworkV1AllowAllToPods
+		expected    []interface{}
+	}{
+		{
+			description: "check for nil allow-all-to-pods recipe network policy ",
+			input:       nil,
+			expected:    nil,
+		},
+		{
+			description: "normal scenario with valid allow-all-to-pods recipe network policy ",
```

Comment:
Since `toPodLabelsKey` is required can you add a test to validate the error scenario?

Reply:
The `FlattenAllowAllToPods` or for that matter any flatten function doesn't return an error in the way they are written. This validation of Required pod labels happens on the terraform schema level. So, even before hitting the API for policies, terraform returns an error in case the user doesn't add this required parameter.